### DISCUSSION TITLE
Confused how the offline db should mirror the server db

### DISCUSSION THREAD

**Discussion by 1mike12** [Nozbe/WatermelonDB#1332] [General]
I totally understand via the examples how WatermelonDB fits in a project that's offline only that sometimes syncs to the cloud. But I have an app that's pretty standard, with users, posts, comments etc. There is a section of infinite scroll that lets users scroll over any public posts. 

1. How would you architect sections of the app that cannot and should not be synced with the local app? IE it makes no sense to synchronize the infinite scroll feed of posts, but they are still obviously the same "type" as the posts you _do_ cache and you would want to share custom methods that you defined on the models. More generically, the subset of locally synced data for a given user must be much smaller than ALL data that exists on a large platform
2. How much do we mirror the server's database structure? For instance, there's obviously only one logged in user client side, so do you still have a users table? Does the currently logged in user exist in this table?

**Comment by KrisLau** [comment]
1. The sync is completely customized by you so you would dictate explicitly which rows/tables you are synchronizing. WatermelonDB basically just gives you a gigantic json of all the changes that have been made to your client side database and you choose what you do with that information,
2. I still keep the users table on the client and yes, it's just basically keeping the server user row record in sync with the client side record.

Not sure if I'm making sense but hopefully that helps!
